doc/man3/flux_future_create.adoc

```diff
@@ -0,0 +1,117 @@
+flux_future_create(3)
+=====================
+:doctype: manpage
+
+
+NAME
+----
+flux_future_create, flux_future_destroy, flux_future_get, flux_future_check, flux_future_then, flux_future_set_timeout - synchronize an activity
+
+
+SYNOPSIS
+--------
+ #include <flux/core.h>
+
+ typedef struct flux_future flux_future_t;
+
+ typedef void (*flux_continuation_f)(flux_future_t *f, void *arg);
+
+ typedef void (*flux_future_init_f)(flux_future_t *f,
+                                    flux_reactor_t *r, void *arg);
+
+ flux_future_t *flux_future_create (flux_future_init_f *cb, void *arg);
```

Comment:
Two use cases:
1) create+get
2) create+then; continuation+get

In the first case, no user provided reactor or callback required.